torchtune/utils/_checkpointing/_checkpointer.py

```diff
@@ -383,6 +383,14 @@ def load_checkpoint(self) -> Dict[str, Any]:
             dim=self._config["hidden_size"],
         )
 
+        if (
+            self._model_type == "GEMMA"
```

Comment:
Hmm so I have a question about this code.

```hf_to_tune``` makes an assumption that ```head_dim * num_heads = dim``` (see [here](https://github.com/pytorch/torchtune/blob/main/torchtune/models/convert_weights.py#L137)). 

But this isn't true for Gemma 7B where ```num_heads=16``` and ```head_dim= 256``` but ```dim=3072``` and not ```4096```. So we will need to differentiate between gemma 2b and 7b here

Reply:
Also, please move to a utility function in checkpointer_utils so we can keep this code clean.

Reply:
If I allow explicit parameter `num_heads ` passing in function `hf_to_tune `, is this allowed?

And I moved them to a utility function in checkpointer_utils.